### PR TITLE
Add catch so errors don't stop the alert loop

### PR DESCRIPTION
**Pull request by JayDwee**

### CODE REVIEW

File: cogs/TwitchAlert.py [hunk 1]
```diff
@@ -522,58 +528,62 @@ async def loop_check_team_live(self):
...
+                        channel = self.bot.get_channel(id=result[0])
+                        try:
+                            # If no Alert is posted
+                            if result[1] is None:
+                                if new_message_embed is None:
+                                    if result[3] is not None:
+                                        message = result[3]
+                                    else:
+                                        message = result[4]
+                                    new_message_embed = await self.create_alert_embed(stream_data, message)
```
Comment (3eak) [inline cogs/TwitchAlert.py:563]: might be worth variabilizing results[0], result[1], result[3] and result[4] to make this more maintainable for the future

File: cogs/TwitchAlert.py [hunk 2]
```diff
@@ -411,54 +411,60 @@ async def loop_check_live(self):
...
+
+                                if new_message_embed is not None:
+                                    new_message = await channel.send(embed=new_message_embed)
+                                    sql_update_message_id = """
+                                    UPDATE UserInTwitchAlert 
+                                    SET message_id = ? 
+                                    WHERE channel_id = ? 
+                                        AND twitch_username = ?"""
+                                    self.ta_database_manager.database_manager.db_execute_commit(
+                                        sql_update_message_id, args=[new_message.id, result[0], current_username])
```
Comment (3eak) [inline cogs/TwitchAlert.py:459]: this looks very familiar to some code further down, could we use a function here to improve maintainability?

File: cogs/TwitchAlert.py [hunk 3]
```diff
@@ -411,54 +411,60 @@ async def loop_check_live(self):
...
-                                sql_update_message_id, args=[new_message.id, result[0], current_username])
-                    except discord.errors.Forbidden as err:
-                        logging.warning(f"TwitchAlert: {err}  Name: {channel} ID: {channel.id}")
-                        sql_remove_invalid_channel = "DELETE FROM TwitchAlerts WHERE channel_id = ?"
-                        self.ta_database_manager.database_manager.db_execute_commit(sql_remove_invalid_channel,
-                                                                                    args=[channel.id])
+            try:
+                if streams_details.get('type') == "live":
+                    current_username = str.lower(streams_details.get("user_name"))
+                    # print(current_username + " is live")
```
Comment (3eak) [inline cogs/TwitchAlert.py:417]: remove (old?) comment

File: cogs/TwitchAlert.py [hunk 4]
```diff
@@ -411,54 +411,60 @@ async def loop_check_live(self):
...
+                                    if result[2] is not None:
+                                        message = result[2]
+                                    else:
+                                        message = result[3]
+
+                                    new_message_embed = await self.create_alert_embed(streams_details, message)
+                                    # with concurrent.futures.ThreadPoolExecutor() as pool3:
+                                    #    new_message = await asyncio.get_event_loop(). \
+                                    #        run_in_executor(pool3, self.create_alert_message, int(result[0]),
+                                    #                        streams_details, message)
```
Comment (3eak) [inline cogs/TwitchAlert.py:449]: remove (old?) comment

File: cogs/TwitchAlert.py [hunk 5]
```diff
@@ -522,58 +528,62 @@ async def loop_check_team_live(self):
...
-                                                                                              current_username])
-                    except discord.errors.Forbidden as err:
-                        logging.warning(f"TwitchAlert: {err}  Name: {channel} ID: {channel.id}")
-                        sql_remove_invalid_channel = "DELETE FROM TwitchAlerts WHERE channel_id = ?"
-                        self.ta_database_manager.database_manager.db_execute_commit(sql_remove_invalid_channel,
-                                                                                    args=[channel.id])
+            try:
+                if stream_data.get('type') == "live":
+                    current_username = str.lower(stream_data.get("user_name"))
+                    # print(current_username + " is live: "+str(usernames))
```
Comment (3eak) [inline cogs/TwitchAlert.py:534]: remove (old?) comment

File: cogs/TwitchAlert.py [hunk 6]
```diff
@@ -522,58 +528,62 @@ async def loop_check_team_live(self):
...
+                                    if result[3] is not None:
+                                        message = result[3]
+                                    else:
+                                        message = result[4]
+                                    new_message_embed = await self.create_alert_embed(stream_data, message)
+
+                                    # with concurrent.futures.ThreadPoolExecutor() as pool3:
+                                    #    new_message = await asyncio.get_event_loop(). \
+                                    #        run_in_executor(pool3, self.create_alert_message, int(result[0]),
+                                    #                        stream_data, message)
```
Comment (3eak) [inline cogs/TwitchAlert.py:568]: remove (old?) comment

**Review by 3eak — CHANGES_REQUESTED** [review]
Just some small things to improve maintanability and clean things up

File: cogs/TwitchAlert.py (hunk 2 above)
Comment (JayDwee) [reply cogs/TwitchAlert.py]: Short answer, yes. Long answer, I am busy but plan to reorganise and test loop_check_live and loop_team_check_live when I am next free.

**Review by 3eak — APPROVED** [review]
LGTM - might be worth adding a TODO or an issue about refactoring those methods though :)

**Comment by JayDwee** [comment]
> LGTM - might be worth adding a TODO or an issue about refactoring those methods though :)

Added to the trello

**Pull request merged**